Beeper/BarcelonaMautrixIPC/Runtime/UnixSocketMautrixIPCChannel.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import Socket
+import Swog
+
+fileprivate let log = Logger(category: "UnixSocketMautrixIPCChannel")
+
+public class UnixSocketMautrixIPCChannel: MautrixIPCInputChannel, MautrixIPCOutputChannel {
+    private let socket: Socket
+    private var readThread: Thread? = nil
+    private var readRunLoop: CFRunLoop? = nil
+    private var readCallback: ((Data) -> ())? = nil
+    
+    public init(_ socketPath: String) {
+        do {
+            try socket = Socket.create(family:.unix)
+            try socket.connect(to: socketPath)
+            log.info("Connected to unix socket \(socketPath)")
+        } catch let error {
+            fatalError("Failed to connect unix socket \(error)")
+        }
+    }
+    
+    private func setupReadThread() {
+        readThread = Thread {
+            self.readRunLoop = CFRunLoopGetCurrent()
+            
+            RunLoop.current.schedule {
+                do {
+                    var shouldKeepRunning = true
```

Comment:
If `shouldKeepRunning` is only used for the `while` below, we can just change the `repeat { } while shouldKeepRunning` to a `repeat { }` (I think) and then get rid of all references to the `shouldKeepRunning` since it'll break out of the loop once we're done listening to the socket.

Reply:
Agree 👍 I'll change it